popde/adaptive_kde.py

```diff
@@ -224,6 +224,147 @@ def optimize_parameters(self, method='loo_cv', fom_plot_name=None):
         return best_params, best_score
 
 
+class KDERescaleOptimization(AdaptiveBwKDE):
+    """
+     A class to optimize rescaling factors for each dimension and 
+      [an optional alpha parameter]
+    using Nelder-Mead optimization. 
+    The optimization is conducted by minimizing a cross-validation
+    score, based on log-likelihood as the figure of merit.
+
+    Attributes:
+        alpha (float): Initial alpha parameter for rescaling.
+        bandwidth (float): Fixed bandwidth, default set to 1.0.
+        n_splits (int): Number of splits for k-fold cross-validation.
+
+    Methods:
+        loo_cv_score(rescale_factors_alpha): 
+            Computes Leave-One-Out cross-validation score 
+            for given rescale factors.
+        kfold_cv_score(rescale_factors_alpha, seed=42): 
+            Computes k-fold cross-validation score 
+            for given rescale factors.
+        optimize_rescale_parameters(initial_rescale_factor, initial_alpha=0.0, method='loo_cv', fom_plot_name=None, bounds=None):
+            Optimizes rescaling factors and alpha parameter.
+    """
+    def __init__(self, data, weights=None, input_transf=None,
+                 stdize=False, rescale=None, backend='KDEpy', bandwidth=1.0, alpha=0.0,
+                 dim_names=None, do_fit=False, n_splits=5):
+        """
+        Initialize the KDERescaleOptimization instance 
+        with data and optional configuration parameters.
+
+        Args:
+            data (array-like): Data to fit the KDE model.
+            weights (array-like, optional): Weights for each data point.
+            input_transf (function, optional): Input transformation function.
+            stdize (bool, optional): Whether to standardize data.
+            rescale (array-like, optional): Initial rescale factors 
+               for each dimension.
+            backend (str, optional): Backend to use, default is 'KDEpy'.
+            bandwidth (float, optional): Fixed bandwidth value, default is 1.0.
+            alpha (float, optional): Initial alpha value, default is 0.0.
+            dim_names (list, optional): Dimension names.
+            do_fit (bool, optional): Whether to fit the KDE upon initialization.
+            n_splits (int, optional): Number of splits for k-fold 
+               cross-validation, default is 2.
```

Comment:
just don't mention the default values and the code will be easier to maintain (here you wrote 2 and then it became wrong)
```suggestion
               cross-validation.
```

Reply:
done